### ISSUE TITLE
Reading line numbers in VS Code

### ISSUE THREAD

**Issue by Qchristensen** [nvaccess/nvda#14268]
Reported by a user via email

### Steps to reproduce:

1. Enable line number reporting in NVDA's document formatting settings.
2. Try to read line numbers in VS Code.

### Actual behavior:

Line numbers are not reported.

### Expected behavior:

Line numbers should be reported when enabled.

#### NVDA version:
NVDA 2022.3

#### Windows version:

Windows 10 x64 build 10.0.19043

#### Name and version of other software in use when reproducing the issue:

Version: 1.72.2 (system setup)
Commit: d045a5eda657f4d7b676dedbfa7aab8207f8a075
Date: 2022-10-12T22:15:18.074Z
Electron: 19.0.17
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
Sandboxed: No

**Comment by rperez030** [comment]
This is a known issue in the Monaco editor. The workaround at this point is to press CTRL +G to get the line number.
For reference: [https://github.com/microsoft/monaco-editor/issues/410](https://github.com/microsoft/monaco-editor/issues/410)

**Comment by leonardder** [comment]
@isidorn may be you could spreat some light on this? is there an issue for this at VS Code's end?

**Comment by isidorn** [comment]
Yes, Rob Perez already pointed to the issue https://github.com/microsoft/monaco-editor/issues/410
We do not know of a good way to solve this, and I am not sure if it would be too noisy if we would read the line number for every line. If there are some suggestions on what to do from the NVDA side do let us know.

fyi @meganrogge

**Comment by leonardder** [comment]
For Visual Studio (not Code), there is a setting that enables line numbers. In the past, that feature always reported line numbers in NVDA, however recently we changed this to respect NVDA's line number setting.
As Code uses a TextArea under the hoot, the best way to make this work for now is look into how Chromium/IA2 exposes line numbers, but I'm afraid it does not.

**Comment by isidorn** [comment]
@leonardder yes, I think the Chromium/IA2 does not expose this.
@meganrogge it could be a setting but it might be too noise for users - since we would use `alert` to read out numbers, which means this could stop / conflict with the reading of the line content. So I think it would make a bad experience.

**Comment by rperez030** [comment]
Visual Studio code currently exposes the line number in the status bar. @leonardder  I wonder if we can use that information to announce line numbers in VS Code when the NVDA setting is on.

**Comment by isidorn** [comment]
We actually also show the line numbers in the status bar. And the status bar has the appropriate label. Not sure if NVDA could fetch it from there.

**Comment by leonardder** [comment]
I guess the best way to fix this would be asking browser vendors to expose the line number in the text attributes on the IAccessibleText interface. However, I'm not sure whether they'd be willing to and whether this would require a spec change.
@jcsteh Would you be able to spreat some light on this?

In the VS code specific case, this would underline the requirement to disable word wrap, though.

**Comment by isidorn** [comment]
@leonardder we actually support word wrap now when a screen reader is on. It is coming in the next VS Code stable release :)
